awx-run/action.yml

```diff
@@ -0,0 +1,117 @@
+name: "Run AWX job or upgrade project"
+description: "Run and wait for an AWX job and or update an AWX project."
+
+inputs:
+  controller-host:
+    description: "AWX server DNS or IP address."
+    required: true
+  controller-username:
+    description: "User name to login."
+    required: true
+  controller-password:
+    description: "Password to login."
+    required: true
+  controller-validate-certs:
+    description: "Validate AWX tls certs. Default is false ."
+    default: "false"
+  project-name:
+    description: "AWX project to update. If not set no project will be updated. Default: empty ."
+  job-template:
+    description: "AWX job to run. If not set no job will be started. Default: empty ."
+  job-extra-vars:
+    description: "Json string with extra vars for job template. Default is a empty json."
+    default: "{}"
+  job-wait-retries:
+    description: "Retries to check if a job still running. Default: is 240 ."
+    default: "240"
+  job-wait-retry-delay:
+    description: "Delay between job wait retries. Default: is 30 ."
+    default: "30"
+  workflow-template:
+    description: "AWX workflow to run. If not set no job will be started. Default: empty ."
+  workflow-node:
+    description: "AWX workflow node to wait for. Default: empty ."
+  workflow-extra-vars:
+    description: "Json string with extra vars for workflow template. Default is a empty json."
+    default: "{}"
+  workflow-timeout:
+    description: "Timeout to wait for an workflow finish in seconds. If not workflow-node is set. Default: is 3600 ."
+    default: "3600"
+  workflow-node-wait-retries:
+    description: "Retries to check if a workflow node still running. Default: is 240 ."
+    default: "240"
+  workflow-node-wait-retry-delay:
+    description: "Delay between workflow node wait retries. Default: is 30 ."
+    default: "30"
+  python-version:
+    description: "Python version to use for running the action. Default is 3.11 ."
+    default: "3.11"
+  skip-installation-on:
+    description: Skip installation on selected runner. Default is self-hosted-generic.
+    default: "self-hosted-generic"
+
+branding:
+  icon: "package"
+  color: "green"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install Python
+      if: ${{ !startsWith(runner.name, inputs.skip-installation-on) }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Install Ansible
+      if: ${{ !startsWith(runner.name, inputs.skip-installation-on) }}
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install --upgrade ansible
```

Comment:
would be better to install it with pipx

Reply:
I have to try it last time i used pipx it didn't link all binary's into /bin.

Reply:
Changed